plugins/tiledb/io/TileDBWriter.cpp

```diff
@@ -252,245 +203,186 @@ void TileDBWriter::initialize()
 
 void TileDBWriter::ready(pdal::BasePointTable& table)
 {
+    // Create objects to store the buffer data.
     auto layout = table.layout();
-    auto all = layout->dims();
-    std::vector<tiledb::Dimension> dims;
+    for (const auto& dimId : layout->dims())
+    {
+        // Get the name and type of the dim.
+        std::string dimName = layout->dimName(dimId);
+        Dimension::Type dimType = layout->dimType(dimId);
+
+        // Allocate the buffer.
+        m_buffers.emplace_back(dimName, dimId, dimType);
+        m_buffers.back().m_buffer.resize(m_args->m_cache_size *
+                                         Dimension::size(dimType));
```

Comment:
You probably need to get the [layout->dimSize](https://github.com/PDAL/PDAL/blob/master/pdal/PointLayout.hpp#L203) because this might not be the same as a default `Dimension::size` if someone added it to the PointLayout differently than expected.